src/manual/task-execution/workers.md

```diff
@@ -0,0 +1,61 @@
+---
+filename: task-execution/workers.md
+title: Workers
+order: 20
+---
+
+Workers are the component responsible for actually executing tasks. Taskcluster
+operates on the "pull" model, where workers pull tasks from queues as they have
+available capacity.
+
+Typically workers are implemented as daemons running on compute instances in
+the cloud. As such, they have no REST API -- that is, there is no "worker
+service".
+
+## Worker Groups and Ids
+
+Each individual worker is identified by a `<workerGroup>/<workerId>` pair.
+This information can be useful to identify errant workers.
+
+## Claiming Tasks
+
+Workers signal their availability for new work by calling the Queue's
+`claimWork` method. If there is a task available, this method returns its
+details and records a "claim" of that task by the calling worker. The claim
+lasts for a short time - a matter of minutes - and must be renewed
+(`reclaimTask`) before it expires. If a claim does expire, the task run is
+marked as `exception` and the task may be automatically retried.
+
+In most cases, multiple workers pull work from the same queue, so the worker
+that executes a particular task is not known in advance.  In practice, this
+does not matter, as all workers with the same `<provisionerId>/<workerType>`
+are typically configured identically.
+
+As mentioned previously, the format of the task payload depends on the worker
+executing the task. It is up to the task author to know the payload format a
+particular worker, as identified by its worker type, will expect.
+
+## Bring Your Own Worker
+
+Workers communicate with the Queue API using normal Taskcluster API calling
+conventions. It is quite practical to run purpose-specific workers with a
+configuration appropriate to that purpose.
```

Comment:
We should add a link to the worker-queue contract of behaviour page, which is required for implementing your own worker.